homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -157,8 +143,10 @@ async def async_update(self) -> None:
                 _state = await self._train_api.async_get_next_train_stop(
                     self._from_station, self._to_station, when
                 )
-        except ValueError as output_error:
-            _LOGGER.error("Departure %s encountered a problem: %s", when, output_error)
+        except ValueError as error:
+            if "Invalid authentication" in error.args[0]:
+                raise ConfigEntryAuthFailed from error
```

Comment:
Raising ConfigEntryAuthFailed won't work here as we're not using a coordinator to fetch new data.

Reply:
I thought it would handle it in the normal update as well but obviously not then.
I have it [here ](https://github.com/home-assistant/core/pull/71432) to move it to coordinator and break out all the entities from attributes (which I didn't think about when I made this PR).
Maybe just close this PR, make the needed changes in the other PR instead and go for that one?

Reply:
I'd just remove raising here and log an error instead for now. The other changes here are good. Then we can merge.